tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -550,8 +550,10 @@ ZTEST_USER(userspace, test_read_other_stack)
 	/* Try to read from another thread's stack. */
 	unsigned int val;
 
-#ifdef CONFIG_MMU
+#if defined(CONFIG_MMU) || defined(CONFIG_MPU)
+#if defined(CONFIG_ARCH_MEM_DOMAIN_SYNCHRONOUS_API)
```

Comment:
Maybe submit this separately?  Seems like this is a coverage problem anyway: if this test can't be run with ARCH_MEM_DOMAIN_SYNCHRONOUS_API=y, then we need to fix it so it can, no?

Maybe I'm just not understanding?  Why does that kconfig force these threads into the same domain to begin with?  The test clearly expects them to be protected from each other, isn't that an API goof if you can turn a kconfig on to break security?

Reply:
This is a side effect of using memory domain, where all threads within one domain can access each others' stack. Hence both read and write tests are skipped with MMU enabled. The Xtensa MPU also supports memory domain. That's why the test needs to be skip. AFAICT, only ARM MPU code changes stack permission during context switches.